content/post/115-ipfs-and-igalia-collaborate-on-dweb-in-browsers.md

```diff
@@ -0,0 +1,195 @@
+
+This problem is actually more general than just local resources. The whole notion of “secure context” is not implemented consistently between browsers, or even within each browser. In addition to the one of [potentially trustworthy origin](https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy) mentioned above, the specification has a slightly more general notion of [potentially trustworthy URL](https://w3c.github.io/webappsec-secure-contexts/#is-url-trustworthy) which used to differentiate from [a priori authenticated URL](https://github.com/w3c/webappsec-mixed-content/issues/35). How these notions should interact with custom handlers is also [currently a bit fuzzy](https://github.com/whatwg/html/issues/5787https://github.com/whatwg/html/issues/5787).
+
+In Chromium, there were [at least 6 implementations of “secure”](https://bugs.chromium.org/p/chromium/issues/detail?id=1153336). Some of these implementations are not very aligned with the current specification. Worse, some of them include historical and non-standard localhost names like `localhost.localdomain`, `localhost6` and `localhost6.localdomain6`. Effort is being made to progressively and carefully unify these implementations and follow the specifications. In particular, patches landed to remove [localhost](https://chromestatus.com/feature/5698580851458048) [names](https://chromestatus.com/feature/5668106045227008) and make more [`data:` URLs potentially trustworthy](https://chromestatus.com/feature/5634194258526208).
+
+## Other fixes
+
+As usual, performing this kind of effort leads to a lot of side tasks: community and specification discussion, code clean up and refactoring, documentation improvement, and other bug fixes. Here are Chromium bugs fixed that are relevant for custom handlers and interoperability with Firefox:
+
+* [Make service workers work with custom handlers](https://chromium-review.googlesource.com/c/chromium/src/+/2487107)
+* [Do not remove %s token when validating (un)registerProtocolHandler’s URL](https://chromium-review.googlesource.com/c/chromium/src/+/2335434)
+* [percent-encode U+0020 SPACE when in URLs computed by custom protocol handlers](https://chromestatus.com/feature/5678518908223488)
+* [percent-encode the delete character when parsing URLs](https://chromestatus.com/feature/5651438652882944)
+
+Finally, for both Chromium and Firefox, [the title argument has been removed from registerProtocolHandler()](https://chromestatus.com/feature/5634386663833600) as per [Mozilla's suggestion](https://github.com/whatwg/html/pull/5425).
+
+If you are interested in an exhaustive list of commits, try expanding the following list:
+
+<details><summary>Chromium</summary>
```

Comment:
Our blog engine does not support collapsed sections. I think we should just show the list, to emphasize how complex and involved this work was:

```suggestion
If you are interested in an exhaustive list of contributions related to this work, here it comes:

###  Chromium
```